pkg/middleware/auth/middleware.go

```diff
@@ -202,3 +230,58 @@ func getReadableCertTypeName(certTypeID string) string {
 	}
 	return certTypeID
 }
+
+// buildResponsePayload creates a BRC-103 response payload according to BRC-104 section 6.9
+func buildResponsePayload(requestID string, statusCode int, headers http.Header, body []byte) ([]byte, error) {
+	writer := util.NewWriter()
+	requestIDBytes, err := base64.StdEncoding.DecodeString(requestID)
+	if err != nil {
+		return nil, fmt.Errorf("invalid request ID format: %w", err)
```

Comment:
```suggestion
		return nil, fmt.Errorf("decode request ID: %w", err)
```
more action base error .. as in above u doing decoding to string

Reply:
To be honest, I like the original err message.

Reply:
I'll stick with my error message if that's just a suggestion.